x-pack/plugins/security/public/management/management_service.ts

```diff
@@ -0,0 +1,110 @@
+ */
+
+import { Subscription } from 'rxjs';
+import { i18n } from '@kbn/i18n';
+import { CoreSetup, FatalErrorsSetup } from 'src/core/public';
+import {
+  ManagementApp,
+  ManagementSetup,
+  ManagementStart,
+} from '../../../../../src/plugins/management/public';
+import { SecurityLicense } from '../../common/licensing';
+import { AuthenticationServiceSetup } from '../authentication';
+import { PluginStartDependencies } from '../plugin';
+import { APIKeysManagementApp } from './api_keys';
+import { RoleMappingsManagementApp } from './role_mappings';
+import { RolesManagementApp } from './roles';
+import { UsersManagementApp } from './users';
+
+interface SetupParams {
+  management: ManagementSetup;
+  license: SecurityLicense;
+  authc: AuthenticationServiceSetup;
+  fatalErrors: FatalErrorsSetup;
+  getStartServices: CoreSetup<PluginStartDependencies>['getStartServices'];
+}
+
+interface StartParams {
+  management: ManagementStart;
+}
+
+export class ManagementService {
+  private license!: SecurityLicense;
+  private licenseFeaturesSubscription?: Subscription;
+
+  setup({ getStartServices, management, authc, license, fatalErrors }: SetupParams) {
+    this.license = license;
+
+    const securitySection = management.sections.register({
+      id: 'security',
+      title: i18n.translate('xpack.security.management.securityTitle', {
+        defaultMessage: 'Security',
+      }),
+      order: 100,
+      euiIconType: 'securityApp',
+    });
+
+    securitySection.registerApp(UsersManagementApp.create({ authc, getStartServices }));
+    securitySection.registerApp(
+      RolesManagementApp.create({ fatalErrors, license, getStartServices })
+    );
+    securitySection.registerApp(APIKeysManagementApp.create({ getStartServices }));
+    securitySection.registerApp(RoleMappingsManagementApp.create({ getStartServices }));
+  }
+
+  start({ management }: StartParams) {
+    this.licenseFeaturesSubscription = this.license.features$.subscribe(async features => {
+      const securitySection = management.sections.getSection('security')!;
+
+      const securityManagementAppsStatuses: Array<[ManagementApp, boolean]> = [
+        [securitySection.getApp(UsersManagementApp.id)!, features.showLinks],
+        [securitySection.getApp(RolesManagementApp.id)!, features.showLinks],
+        [securitySection.getApp(APIKeysManagementApp.id)!, features.showLinks],
+        [
+          securitySection.getApp(RoleMappingsManagementApp.id)!,
+          features.showLinks && features.showRoleMappingsManagement,
+        ],
+      ];
+
+      // Iterate over all registered apps and update their enable status depending on the available
+      // license features.
+      for (const [app, enableStatus] of securityManagementAppsStatuses) {
+        if (app.enabled === enableStatus) {
+          continue;
+        }
+
+        if (enableStatus) {
+          app.enable();
+        } else {
+          app.disable();
```

Comment:
This is probably a bug or design choice in the `management` plugin, but disabling our apps does not prevent routing to them. It merely hides the link from the management side nav. If you know the link to a disabled app, or if another app links you to a disabled app, it will still render as if it's enabled. @mattkime do you know if this is intentional or not?

@azasypkin If this is intentional, then I think we still need some version of the `checkLicense` toast message below that you asked about.

Reply:
Hmm, good catch! Having `disable` that just hides doesn't feel right to me, it must be a bug or the wrong method name...

Reply:
I'll help with this. This is a case where the solution during and after the transition to the new api is different.